ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -781,35 +813,38 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
                   View Preview Table
               </Clickable>
             </div>
-            {previewDataLoading && <div style={styles.warningMessage}>
-              <Spinner style={{position: 'relative', top: '2rem'}} />
-              <div style={{top: '3rem', position: 'relative'}}>
-                It may take up to few minutes to load the data
-              </div>
-            </div>}
             {previewList.length > 0 &&
               <div style={{display: 'flex', flexDirection: 'column'}}>
                 <div style={{display: 'flex', flexDirection: 'row', paddingTop: '0.5rem'}}>
                   {previewList.map(previewRow =>
-                     <Clickable key={previewRow.domain}
-                          onClick={() =>
-                            this.setState({selectedPreviewDomain: previewRow.domain})}
-                            style={{
-                              marginLeft: '0.2rem', color: colors.accent,
-                              marginRight: '0.25rem', paddingBottom: '0.25rem',
-                              width: '7rem',
-                              borderBottom:
-                               (selectedPreviewDomain === previewRow.domain) ?
-                                 '4px solid ' + colors.accent : '',
-                              fontWeight: (selectedPreviewDomain === previewRow.domain)
-                               ? 600 : 400,
-                              fontSize: '18px',
-                              display: 'flex',
-                              justifyContent: 'center',
-                              lineHeight: '32px',
-                            }}>
-                       {previewRow.domain === 'OBSERVATION' ? 'SURVEY' : previewRow.domain}
-                     </Clickable>
+                     <TooltipTrigger content={'Preview for domain ' + previewRow.domain.toString()
+                     + ' is still loading. It may take up to one minute'}
+                                     disabled={!previewRow.isLoading}
+                                     side='top'>
+                       <Clickable key={previewRow.domain.toString()}
+                            disabled={previewRow.isLoading}
+                            onClick={() =>
+                              this.setState({selectedPreviewDomain: previewRow.domain})}
+                              style={{
+                                marginLeft: '0.2rem', color: colors.accent,
+                                marginRight: '0.25rem', paddingBottom: '0.25rem',
```

Comment:
More than 3 attributes, can we move this to style

Reply:
Will move.